models/taskRequests.js

```diff
@@ -10,30 +10,43 @@ const userModel = require("./users");
  *
  * @return {Object}
  */
-const fetchTaskRequests = async () => {
+const fetchTaskRequests = async (dev) => {
   const taskRequests = [];
-
+  const newTaskRequestsModel = [];
   try {
     const taskRequestsSnapshots = (await taskRequestsCollection.get()).docs;
 
     const taskPromises = [];
     const userPromises = [];
 
+    const newUserPromises = [];
+
     taskRequestsSnapshots.forEach((taskRequestsSnapshot) => {
       const taskRequestData = taskRequestsSnapshot.data();
       taskRequestData.id = taskRequestsSnapshot.id;
       taskRequestData.url = new URL(`/taskRequests/${taskRequestData.id}`, config.get("services.rdsUi.baseUrl"));
       const { requestors } = taskRequestData;
 
-      taskPromises.push(tasksModel.fetchTask(taskRequestData.taskId));
-      userPromises.push(Promise.all(requestors.map((requestor) => userModel.fetchUser({ userId: requestor }))));
-
-      taskRequests.push(taskRequestData);
+      if (taskRequestData.taskId && !taskRequestData.requestType) {
+        taskPromises.push(tasksModel.fetchTask(taskRequestData.taskId));
+        userPromises.push(Promise.all(requestors.map((requestor) => userModel.fetchUser({ userId: requestor }))));
+        taskRequests.push(taskRequestData);
+      } else if (dev) {
+        newUserPromises.push(Promise.all(requestors.map((requestor) => userModel.fetchUser({ userId: requestor }))));
+        newTaskRequestsModel.push(taskRequestData);
+      }
     });
 
     const tasks = await Promise.all(taskPromises);
     const users = await Promise.all(userPromises);
 
+    if (dev) {
+      const newUsers = await Promise.all(newUserPromises);
+      newTaskRequestsModel.forEach((taskRequest, index) => {
+        taskRequest.requestors = newUsers[+index];
```

Comment:
- should we change the name here from `requestors` to `requests` as these are task requests?